akvo/rsr/filters.py

```diff
@@ -98,26 +105,47 @@ def get_locations(location, locations):
 def location_choices(qs):
     """From a queryset get possible location filter choices"""
 
+    country_ids = get_location_country_ids(qs)
+    if qs.model is Project:
+        country_ids += get_recipient_country_ids(qs)
+
+    locations = [
```

Comment:
This part I don't even understand, but doint locations = list(set(locations)) at line 117 should also shorten the list